lib/runtime.cpp

```diff
@@ -495,11 +497,12 @@ int ModuleLoader::initialize() {
             return everest.telemetry_publish(category, subcategory, type, telemetry);
         };
 
+        module_adapter.get_mapping = [&everest]() { return everest.get_3_tier_model_mapping(); };
+
         this->callbacks.register_module_adapter(module_adapter);
 
         // FIXME (aw): would be nice to move this config related thing toward the module_init function
-        std::vector<cmd> cmds =
-            this->callbacks.everest_register(config.get_main_config()[this->module_id]["connections"]);
+        std::vector<cmd> cmds = this->callbacks.everest_register(config.get_fulfillments(this->module_id));
```

Comment:
I just saw that now, why do we pass the fulfillments via `everest_register`?  `everest_register` wasn't intended to transfer this kind of information to the module - it's intend was to get the implemented commands of the module (if I remember correctly).
This information should be passed via the init function.  If possible, I would add this information to the `ModuleInfo` struct, in case it can be consumed directly via user code, or if not pass it as an additional argument and then it's consumption needs to take place in the `ld-ev` code which finds the corresponding requirement and sets its `_mapping` from that.